### PR TITLE
adding fake ip for pod from core

### PR DESCRIPTION
**Pull request by shireenf-ibm**

### CODE REVIEW

**Comment by adisos** [comment]
please read this issue, as it is relevant: https://github.com/np-guard/netpol-analyzer/issues/44
does your change assign fake ip only to pods read from files?

**Comment by shireenf-ibm** [comment]
> please read this issue, as it is relevant: #44 does your change assign fake ip only to pods read from files?

not sure, will re-check and update

**Review by adisos — CHANGES_REQUESTED** [review]

File: pkg/netpol/eval/internal/k8s/pod.go [hunk 1]
```diff
@@ -210,7 +207,3 @@ func namespacedName(pod *corev1.Pod) string {
 func variantFromLabelsMap(labels map[string]string) string {
 	return hex.EncodeToString(sha1.New().Sum([]byte(fmt.Sprintf("%v", labels)))) //nolint:gosec
 }
-
-func getFakePodIP() string {
-	return ipv4LoopbackAddr
```
[the remark this answers is not in the document]
Comment (shireenf-ibm) [reply pkg/netpol/eval/internal/k8s/pod.go:215]: done

File: pkg/netpol/eval/internal/k8s/pod.go [hunk 2]
```diff
@@ -188,7 +185,7 @@ func PodsFromWorkloadObject(workload interface{}, kind string) ([]*Pod, error) {
 		pod.Labels = make(map[string]string, len(podTemplate.Labels))
 		pod.IPs = make([]corev1.PodIP, 0)
 		pod.Ports = make([]corev1.ContainerPort, 0, defaultPortsListSize)
-		pod.HostIP = getFakePodIP()
```
[the remark this answers is not in the document]
Comment (shireenf-ibm) [reply pkg/netpol/eval/internal/k8s/pod.go:191]: done

File: pkg/netpol/eval/internal/k8s/pod.go [hunk 3]
```diff
@@ -57,7 +54,7 @@ type Owner struct {
 
 // PodFromCoreObject creates a PodRef by extracting relevant fields from the k8s Pod
 func PodFromCoreObject(p *corev1.Pod) (*Pod, error) {
-	if p.Status.HostIP == "" || len(p.Status.PodIPs) == 0 { // not scheduled nor assigned IP addresses - ignore
+	if p.Status.HostIP == "" && len(p.Status.PodIPs) == 0 { // not scheduled nor assigned IP addresses - ignore
```
[the remark this answers is not in the document]
Comment (adisos) [reply pkg/netpol/eval/internal/k8s/pod.go:57]: `HostIP` is IP address of the host to which the pod is assigned. Empty if not yet scheduled.
`PodIPs` holds the IP addresses allocated to the pod. If this field is specified, the 0th entry must match the podIP field.(which is IP address allocated to the pod.Empty if not yet allocated.).

I think we need both. on a live cluster we should ignore such pods (if at least one condition holds).
Let's revert this change, and also add in the scanner a fake pod ip to `p.Status.PodIPs`.

[the remark this answers is not in the document]
Comment (shireenf-ibm) [reply pkg/netpol/eval/internal/k8s/pod.go]: - Is assigning the fake ip to the `p.Status.PodIPs[0]` enough? should check if there is a` p.Status.PodIP`? 
- Is it okay to use same fake ip for both `p.Status.PodIPs[0] `and` p.Status.HostIP`?

[the remark this answers is not in the document]
Comment (adisos) [reply pkg/netpol/eval/internal/k8s/pod.go]: - yes, and you should only check the slice `p.Status.PodIPs`.
- yes

File: pkg/netpol/scan/scan.go [hunk 4]
```diff
@@ -26,6 +26,9 @@ import (
 	"github.com/np-guard/netpol-analyzer/pkg/netpol/logger"
 )
 
+// IPv4LoopbackAddr is used as fake IP in the absenceof Pod.Status.HostIP or Pod.Status.PodIPs
```
Comment (adisos) [inline pkg/netpol/scan/scan.go:29]:
```suggestion
// IPv4LoopbackAddr is used as fake IP in the absence of Pod.Status.HostIP or Pod.Status.PodIPs
```

File: pkg/netpol/scan/scan.go [hunk 5]
```diff
@@ -401,11 +404,21 @@ func convertPodListTOK8sObjects(pl *v1.PodList) ([]K8sObject, error) {
 		if isValidKind, err := validateNamespaceAndKind(&pl.Items[i].Namespace, &pl.Items[i].Kind, Pod); !isValidKind {
 			return nil, err
 		}
+		checkAndUpdatePodStatusIPsFields(&pl.Items[i])
 		res[i] = K8sObject{Pod: &pl.Items[i], Kind: Pod}
 	}
 	return res, nil
 }
 
+func checkAndUpdatePodStatusIPsFields(rc *v1.Pod) {
```
Comment (adisos) [inline pkg/netpol/scan/scan.go:413]:
```suggestion
// checkAndUpdatePodStatusIPsFields adds fake IP to pod.Status.HostIP or pod.Status.PodIPs if missing
func checkAndUpdatePodStatusIPsFields(rc *v1.Pod) {
```

**Review by adisos — APPROVED** [review]

**Pull request merged**